.github/workflows-source/release-testing.yml

```diff
@@ -182,3 +182,41 @@ jobs:
         with:
           <<: *bazel-bep-with
           name: ${{ github.job }}-${{ matrix.version }}-bep
+
+  repro-checks:
+    name: Repro check using mirror ${{ matrix.mirror }} of `${{ matrix.os }}`
+    # `ci-main` is the job that pushes the images to the cloud storage
+    needs:
+      - ci-main
+    strategy:
+      # This should spawn 6 jobs, all combinations of `mirror` x `os`
+      matrix:
+        mirror:
+          - "https://download.dfinity.systems"
+          - "https://download.dfinity.network"
+        os:
+          - --guestos
+          - --setupos
+          - --hostos
+    <<: *dind-large-setup
+    steps:
+      - name: Setup curl 8.3.0
+        shell: bash
+        run: |
+          curl -LO https://curl.se/download/curl-8.3.0.tar.gz
+          tar -xzf curl-8.3.0.tar.gz
+          mkdir ~/curl
+          cd curl-8.3.0
+          ./configure --with-ssl --prefix=/home/ubuntu/curl/
+          make -j$(nproc)
+          sudo make install
```

Comment:
I would rather upgrade `curl` version in the dev container image since right now it has the version `7.68` but in the `repro-check.sh` script we use a feature introduced in version `7.73` `--output-dir`